src/Credentials.js

```diff
@@ -233,4 +245,25 @@ const configNetworks = (nets) => {
   return nets
 }
 
+const padMessage = (message) => {
+  const INTERVAL_LENGTH = 50
+  const padLenght = INTERVAL_LENGTH - message.length % INTERVAL_LENGTH
+
+  return message + ' '.repeat(padLenght)
+}
+
+const encryptMessage = (message, receiverKey) => {
+  const tmpKp = nacl.box.keyPair()
```

Comment:
what this key pair used for? just because nacl also authenticates messages as well? is it signing it with this key? Seems to make the payload a lot larger to have to include an additional pub key and signature

Reply:
The temp keyPair is needed to do a diffie-hellman key exchange. The message can only be decrypted if the mobile app has the tmpPubKey and it's own secretKey.